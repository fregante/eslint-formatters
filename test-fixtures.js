// Mock ESLint results for testing formatters

// No errors
export const noErrors = [];

// Single file with multiple errors (errors and warnings)
export const singleFile = [
	{
		filePath: 'foo.js',
		messages: [
			{
				message: 'Unexpected foo.',
				severity: 2,
				line: 5,
				column: 10,
				ruleId: 'foo',
			},
			{
				message: 'Missing semicolon.',
				severity: 2,
				line: 10,
				column: 15,
				ruleId: 'semi',
			},
			{
				message: 'Unused variable.',
				severity: 1,
				line: 3,
				column: 8,
				ruleId: 'no-unused-vars',
			},
		],
		errorCount: 2,
		warningCount: 1,
		fixableErrorCount: 1,
		fixableWarningCount: 0,
	},
];

// Multiple files with different error types
export const multipleFiles = [
	{
		filePath: 'foo.js',
		messages: [
			{
				message: 'Unexpected foo.',
				severity: 2,
				line: 5,
				column: 10,
				ruleId: 'foo',
			},
		],
		errorCount: 1,
		warningCount: 0,
		fixableErrorCount: 0,
		fixableWarningCount: 0,
	},
	{
		filePath: 'bar.js',
		messages: [
			{
				message: 'Unexpected bar.',
				severity: 1,
				line: 6,
				column: 11,
				ruleId: 'bar',
			},
		],
		errorCount: 0,
		warningCount: 1,
		fixableErrorCount: 0,
		fixableWarningCount: 0,
	},
	{
		filePath: 'baz.js',
		messages: [
			{
				message: 'Fatal error parsing file.',
				severity: 2,
				fatal: true,
				line: 1,
				column: 1,
				ruleId: null,
			},
		],
		errorCount: 1,
		warningCount: 0,
		fixableErrorCount: 0,
		fixableWarningCount: 0,
	},
];

// Multiple errors and warnings with fixable issues
export const withFixableIssues = [
	{
		filePath: 'fixable.js',
		messages: [
			{
				message: 'Missing semicolon.',
				severity: 2,
				line: 5,
				column: 20,
				ruleId: 'semi',
			},
			{
				message: 'Unexpected trailing comma.',
				severity: 2,
				line: 8,
				column: 15,
				ruleId: 'comma-dangle',
			},
			{
				message: 'Prefer const over let.',
				severity: 1,
				line: 3,
				column: 1,
				ruleId: 'prefer-const',
			},
		],
		errorCount: 2,
		warningCount: 1,
		fixableErrorCount: 2,
		fixableWarningCount: 1,
	},
];
