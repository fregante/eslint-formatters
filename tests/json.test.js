import {describe, it, expect} from 'vitest';
import formatter from '../packages/eslint-formatter-json/index.js';
import {mockResults} from '../test-fixtures.js';

describe('eslint-formatter-json', () => {
	it('should format results correctly', () => {
		const output = formatter(mockResults);
		expect(output).toMatchSnapshot();
	});

	it('should return empty array for no errors', () => {
		const output = formatter([]);
		expect(output).toBe('[]');
	});
});
