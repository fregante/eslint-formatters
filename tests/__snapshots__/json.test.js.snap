// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`eslint-formatter-json > should format results correctly 1`] = `"[{"filePath":"foo.js","messages":[{"message":"Unexpected foo.","severity":2,"line":5,"column":10,"ruleId":"foo"}],"errorCount":1,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"bar.js","messages":[{"message":"Unexpected bar.","severity":1,"line":6,"column":11,"ruleId":"bar"}],"errorCount":0,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0}]"`;
